src/libsyntax/parse/parser.rs

```diff
@@ -4768,7 +4768,8 @@ impl<'a> Parser<'a> {
                 } else {
                     let mut err = self.diagnostic()
                         .struct_span_err(sp, "can't qualify macro invocation with `pub`");
-                    err.help("try adjusting the macro to put `pub` inside the invocation");
+                    err.help(concat!("try adjusting the macro to put `pub` inside the invocation.",
```

Comment:
btw you can do the same without concat!, but by doing "text text text \
            text text".